### ISSUE TITLE
Mysql Logger

### ISSUE THREAD

**Issue by nevodavid** [typeorm/typeorm#555]
Hi :)
I'm using the ORM logger in my project **logger: (level: string, message: any)**,
This thing is fantastic ^^
Is there anyway that I can pass this thing the user request object?
Maybe new feature in the future?

**Comment by pleerock** [comment]
its quite problematic to implement. because you need request-based object in a very global and far-away place.

**Comment by nevodavid** [comment]
We can start by adding option to add meta data parameter to `this.query()` ?

**Comment by pleerock** [comment]
its not enough. `query` is executed by various of methods including persist remove and query builder and in all those methods you need meta parameters as well to pass it to `query` then.

**Comment by pleerock** [comment]
if I point you direction will you able to implement this feature?

**Comment by nevodavid** [comment]
Well actually it's enough for me to use it only when I use the "query" so if I can add one optional parameter it will be enough, what's are your suggestions?

**Comment by pleerock** [comment]
In latest commits I have updated how logger works. Now what you want can be implemented this way:

```ts
// user sends request during entity save
postRepository.save(post, { data: { request: request } });

// in custom logger class you can access it this way:
logQuery(query: string, parameters?: any[], queryRunner?: QueryRunner) {
    const requestUrl = queryRunner && queryRunner.data["request"] ? "(" + queryRunner.data["request"].url + ") " : "";
    console.log(requestUrl + "executing query: " + sql);
}
```

More info about new logger [here](https://github.com/typeorm/typeorm/blob/master/docs/logging.md).

Changes will be released in 0.1.0-alpha.33 version.

**Comment by nevodavid** [comment]
you are the best :)
